crates/bindings-csharp/Codegen/Module.cs

```diff
@@ -117,21 +175,122 @@ public TableDeclaration(GeneratorAttributeSyntaxContext context)
+        Views = new(context.Attributes.Select(a => new TableView(this, a)).ToImmutableArray());
 
-        if (Scheduled is not null)
+        var schedules = Views.Select(t => t.Scheduled);
+        var numSchedules = schedules.Count();
+        if (numSchedules > 0)
+        {
+            var distinctSchedules = schedules.Distinct();
+            if (numSchedules != Views.Length || distinctSchedules.Count() != 1)
+            {
+                throw new Exception(
+                    "When using multiple [Table] attributes with schedule, all [Table] must have the same schedule."
+                );
+            }
+
+            Scheduled = distinctSchedules.First();
+            if (Scheduled != null)
+            {
+                // For scheduled tables, we append extra fields early in the pipeline,
+                // both to the type itself and to the BSATN information, as if they
+                // were part of the original declaration.
+                Members = new(Members.Concat(ScheduledColumns(FullName)).ToImmutableArray());
+            }
+        }
+    }
+
+    protected override ColumnDeclaration ConvertMember(IFieldSymbol field) => new(FullName, field);
+
+    public IEnumerable<string> GenerateViewFilters(string viewName, string iTable)
+    {
+        foreach (
+            var (f, i) in Members
+                .Select((field, i) => (field, i))
+                .Where(pair => pair.field.IsEquatable)
+        )
         {
-            // For scheduled tables, we append extra fields early in the pipeline,
-            // both to the type itself and to the BSATN information, as if they
-            // were part of the original declaration.
-            Members = new(Members.Concat(ScheduledColumns).ToImmutableArray());
+            var globalName = $"global::{FullName}";
+            var colEqWhere = $"{iTable}.ColEq.Where({i}, {f.Name}, {globalName}.BSATN.{f.Name})";
+
+            yield return $"""
+                public IEnumerable<{globalName}> FilterBy{f.Name}({f.Type} {f.Name}) =>
+                    {colEqWhere}.Iter();
+                """;
+
+            if (f.GetAttrs(viewName).HasFlag(ColumnAttrs.Unique))
+            {
+                yield return $"""
+                    public {globalName}? FindBy{f.Name}({f.Type} {f.Name}) =>
+                        FilterBy{f.Name}({f.Name})
+                        .Cast<{globalName}?>()
+                        .SingleOrDefault();
+
+                    public bool DeleteBy{f.Name}({f.Type} {f.Name}) =>
+                        {colEqWhere}.Delete();
+
+                    public bool UpdateBy{f.Name}({f.Type} {f.Name}, {globalName} @this) =>
+                        {colEqWhere}.Update(@this);
+                    """;
+            }
         }
     }
 
-    protected override ColumnDeclaration ConvertMember(IFieldSymbol field) => new(field);
+    public IEnumerable<
+        KeyValuePair<(string viewName, string tableName), (string view, string getter)>
```

Comment:
Nit - it doesn't look like this is even used as actual key-values, but rather each field has its own meaning... so can it be just

```suggestion
        (string viewName, string tableName, string view, string getter)
```

instead to reduce nesting?

Reply:
Also I'd still suggest extracting it into `record struct TableView`.

Unlike type fields, tuple names are compile-time-only syntax sugar, they don't even show up in reflection, and `(string Item1, string Item2)` is rather unhelpful to look at in a debugger or in any serialization context.